src/fs/driver/fat/fat_common.c

```diff
@@ -0,0 +1,1608 @@
+/**
+ * @file   fat_common.c
+ * @brief  VFS-independent part of FAT driver
+ * @data   9 Apr 2015
+ * @author Denis Deryugin
+ */
+
+#include <assert.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <limits.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <embox/block_dev.h>
+#include <framework/mod/options.h>
+#include <fs/fat.h>
+#include <mem/misc/pool.h>
+#include <util/math.h>
+
+#define LABEL "EMBOX_DISK"
+#define SYSTEM "FAT12"
```

Comment:
Yes, current FAT driver is only able to format drive to FAT12.
However, it can mount and use FAT12/16/32 if it was formatted before.
Implementing FAT16/32 formatting is added to Trello, but it's not very important right now.